drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -0,0 +1,849 @@
+				   GPIO_DIR_IN | GPIO_PUD_NORMAL);
+	}
+
+	return 0;
+}
+
+static inline int configure_spi(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	const struct rf2xx_config *conf = dev->config->config_info;
+
+	/* Get SPI Driver Instance*/
+	ctx->spi = device_get_binding(conf->spi.devname);
+	if (!ctx->spi) {
+		LOG_ERR("Failed to get instance of %s device",
+			conf->spi.devname);
+
+		return -ENODEV;
+	}
+
+	/* Apply SPI Config: 8-bit, MSB First, MODE-0 */
+	ctx->spi_cfg.operation = SPI_WORD_SET(8) |
+				 SPI_TRANSFER_MSB;
+	ctx->spi_cfg.slave = conf->spi.addr;
+	ctx->spi_cfg.frequency = conf->spi.freq;
+	ctx->spi_cfg.cs = NULL;
+
+	/*
+	 * Get SPI Chip Select Instance
+	 *
+	 * This is an optinal feature configured on DTS. Some SPI controllers
+	 * automatically set CS line by device slave address. Check your SPI
+	 * device driver to understand if you need this option enabled.
+	 */
+	ctx->spi_cs.gpio_dev = device_get_binding(conf->spi.cs.devname);
+	if (ctx->spi_cs.gpio_dev) {
+		ctx->spi_cs.gpio_pin = conf->spi.cs.pin;
+		ctx->spi_cs.delay = 0U;
+
+		ctx->spi_cfg.cs = &ctx->spi_cs;
+
+		LOG_DBG("SPI GPIO CS configured on %s:%u",
+			conf->spi.cs.devname, conf->spi.cs.pin);
+	}
+
+	return 0;
+}
+
+static int rf2xx_init(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	const struct rf2xx_config *conf = dev->config->config_info;
+	char thread_name[20];
+
+	LOG_DBG("\nInitialize RF2XX Transceiver\n");
+
+	k_mutex_init(&ctx->phy_mutex);
+	k_sem_init(&ctx->trx_tx_sync, 0, 1);
+	k_sem_init(&ctx->trx_isr_lock, 0, 1);
+	k_timer_init(&ctx->trx_isr_timeout, trx_isr_timeout, NULL);
+	k_timer_user_data_set(&ctx->trx_isr_timeout, ctx);
+
+	if (configure_gpios(dev) != 0) {
+		LOG_ERR("Configuring GPIOS failed");
+		return -EIO;
+	}
+
+	if (configure_spi(dev) != 0) {
+		LOG_ERR("Configuring SPI failed");
+		return -EIO;
+	}
+
+	LOG_DBG("GPIO and SPI configured");
+
+	if (power_on_and_setup(dev) != 0) {
+		LOG_ERR("Configuring RF2XX failed");
+		return -EIO;
+	}
+
+	k_thread_create(&ctx->trx_thread,
+			ctx->trx_stack,
+			CONFIG_IEEE802154_RF2XX_RX_STACK_SIZE,
+			(k_thread_entry_t) rf2xx_thread_main,
+			dev, NULL, NULL,
+			K_PRIO_COOP(2), 0, K_NO_WAIT);
+
+	sprintf(thread_name, "802.15.4 main [%d]", conf->inst);
+	k_thread_name_set(&ctx->trx_thread, thread_name);
+
+	return 0;
+}
+
+static void rf2xx_iface_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct rf2xx_context *ctx = dev->driver_data;
+	u8_t *mac = get_mac(dev);
+
+	net_if_set_link_addr(iface, mac, 8, NET_LINK_IEEE802154);
+
+	ctx->iface = iface;
+
+	ieee802154_init(iface);
+}
+
+static struct ieee802154_radio_api rf2xx_radio_api = {
+	.iface_api.init   = rf2xx_iface_init,
+
+	.get_capabilities = rf2xx_get_capabilities,
+	 /**
+	  *  CCA is not recommended to be started manually when using
+	  * extended mode.
+	  * Datasheet: chapter 8.6 Clear Channel Assessment (CCA)
+	  */
+	.cca              = NULL,
```

Comment:
cca function is not about starting cca, it's about getting the cca status. 


Your driver is only working with aloha radio driver then, which is not the default one. 

Please fix this: provide a way to get the cca so csma-ca radio driver is going to work.

Reply:
This radio will perform automatically CSMA/CA on HW FSM when TX is started . In this case I can not get real CCA status before start transmission. So, a dummy function to return free channel is enough?

Reply:
yep, have a function that always returns 0. That will do

Reply:
Done!